src/agentscope/agents/react_agent.py

```diff
@@ -129,13 +94,22 @@ def __init__(
         self.memory.add(Msg("system", self.sys_prompt, role="system"))
 
         # Initialize a parser object to formulate the response from the model
-        self.parser = MarkdownJsonDictParser(
-            content_hint={
-                "thought": "what you thought",
-                "speak": "what you speak",
-                "function": service_toolkit.tools_calling_format,
-            },
-            required_keys=["thought", "speak", "function"],
+        self.parser = RegexTaggedContentParser(
```

Comment:
Instead of just remove the former `MarkdownJsonDictParser`, should we give an argument to user, and let the user choose which parser they want? e.g. (parser: Literal["json", "tag"] = "tag")